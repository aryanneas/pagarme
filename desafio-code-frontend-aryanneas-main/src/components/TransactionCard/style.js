import styled from 'styled-components'

export const TransactionWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid #f2f2f3;
  padding: 0 5%;
  &:last-child {
    border-bottom: none;
  }
`

export const ClientInfo = styled.div`
  color: #454550;
`

export const Name = styled.p`
  font-weight: bold;
`

export const TransactionInfo = styled.div`
  text-align: right;
`

export const Status = styled.p`
  font-size: 14px;
  color: #72737a;
`
export const Value = styled.p`
  color: #2b2b2b;
  font-weight: bold;
`
