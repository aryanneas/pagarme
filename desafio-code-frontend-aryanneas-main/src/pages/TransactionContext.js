import React from 'react'

export default React.createContext({
  transactions: null,
  setTransactions: () => {},
})
