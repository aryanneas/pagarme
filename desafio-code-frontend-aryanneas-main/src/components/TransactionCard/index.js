import React from 'react'

import {
  formatStatus,
  formatDateTime,
  formatCurrency,
} from '../../utils/formatting'

import {
  TransactionWrapper,
  ClientInfo,
  Name,
  TransactionInfo,
  Status,
  Value,
} from './style'

import TransactionContext from '../../pages/TransactionContext'

const TransactionCard = () => {
  return (
    <TransactionContext.Consumer>
      {(value) =>
        value.transactions
          ? value.transactions.map((transaction) => (
              <TransactionWrapper key={transaction.id}>
                <ClientInfo>
                  <Name>{transaction.credit_card_holder_name}</Name>
                  <p>{formatDateTime(transaction.date)}</p>
                </ClientInfo>

                <TransactionInfo>
                  <Status>{formatStatus(transaction.status)}</Status>
                  <Value>{formatCurrency(transaction.amount)}</Value>
                </TransactionInfo>
              </TransactionWrapper>
            ))
          : null
      }
    </TransactionContext.Consumer>
  )
}

export default TransactionCard
