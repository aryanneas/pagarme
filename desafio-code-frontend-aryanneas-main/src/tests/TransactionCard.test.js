import React from 'react'
import { render } from '@testing-library/react'

import TransactionContext from '../pages/TransactionContext'
import TransactionCard from '../components/TransactionCard'

const ComponentConsumer = () => (
  <TransactionContext.Consumer>
    {() => <TransactionCard />}
  </TransactionContext.Consumer>
)

describe('TransactionCard page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ComponentConsumer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
