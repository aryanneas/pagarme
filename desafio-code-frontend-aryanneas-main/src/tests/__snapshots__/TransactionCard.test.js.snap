// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionCard page should render correctly 1`] = `<DocumentFragment />`;
